src/core.js

```diff
@@ -85,25 +90,32 @@ Promise.prototype._handle = function(deferred) {
   var state = this._state;
   var value = this._value;
   asap(function() {
-    var cb = state === 1 ? deferred.onFulfilled : deferred.onRejected
+    var cb = state === 1 ? deferred.onFulfilled : deferred.onRejected;
     if (cb === null) {
-      (state === 1 ? deferred.promise._resolve(value) : deferred.promise._reject(value))
-      return
+      if (state === 1) {
+        deferred.promise._resolve(value);
+      } else {
+        deferred.promise._reject(value);
+      }
+      return;
     }
     var ret = tryCallOne(cb, value);
     if (ret === IS_ERROR) {
-      deferred.promise._reject(LAST_ERROR)
+      deferred.promise._reject(LAST_ERROR);
     } else {
-      deferred.promise._resolve(ret)
+      deferred.promise._resolve(ret);
     }
   });
 };
 Promise.prototype._resolve = function(newValue) {
-  //Promise Resolution Procedure: https://github.com/promises-aplus/promises-spec#the-promise-resolution-procedure
+  // Promise Resolution Procedure: https://github.com/promises-aplus/promises-spec#the-promise-resolution-procedure
   if (newValue === this) {
-    return this._reject(new TypeError('A promise cannot be resolved with itself.'))
+    return this._reject(
+      new TypeError('A promise cannot be resolved with itself.')
+    );
   }
-  if (newValue && (typeof newValue === 'object' || typeof newValue === 'function')) {
+  var type = (newValue && typeof newValue);
+  if (type === 'object' || type === 'function') {
```

Comment:
This changes the actual code and is likely to have a performance impact.  Many JS implementations will treat `typeof foo === 'function'` as "Check `foo` is a function" but treat `var t = typeof foo; t === 'function'` as "Get the type of `'foo'` as a string, compare the two strings `t` and `'function'`".  I haven't actually benchmarked this in this particular instance though.

I think it is also clearer to see this as "`newValue` is truthy and of type object or function", I can't come up with an obvious english language description of your code.